lib/puppet/face/config.rb

```diff
@@ -159,6 +159,9 @@ def report_section_and_environment(section_name, environment_name)
         report_section_and_environment(options[:section], Puppet.settings[:environment])
       end
 
+      # Try to set the value to validate it
+      Puppet.settings[name.to_sym] = value
```

Comment:
Should we only validate a subset of settings eg `certname`? Since `BaseSetting#validation` is broken, should we just call settings with hooks?